### PR TITLE
13737 More Vue cleanup

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* bcgov/entity#13737

*Description of changes:*

- deleted obsolete declaration files
- imported Vue from vue not vue-property-decorator
- changed slot="label" to v-slot:label
- fixed a mixin declaration
- deleted a mixin and moved function to component that uses it
- updated Storybook

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

File: src/components/staff-comments/StaffComments.vue [hunk 1]
```diff
@@ -203,6 +203,22 @@ export default class StaffComments extends Mixins(DateMixin, FilingMixin) {
...
+  private flattenAndSortComments (comments: Array<any>): Array<any> {
+    if (comments && comments.length > 0) {
+      // first use map to change comment.comment to comment
+      const temp: Array<any> = comments.map(c => c.comment)
+      // then sort newest to oldest
+      temp.sort((a, b) => new Date(a.timestamp) < new Date(b.timestamp) ? 1 : -1)
+      return temp
+    }
+    return []
+  }
```
Comment (severinbeauvais) [inline src/components/staff-comments/StaffComments.vue:221]: This is the only component that used this function, so I moved it here.

File: src/mixins/filing-mixin.ts [hunk 2]
```diff
@@ -1,23 +0,0 @@
-import { Component, Vue } from 'vue-property-decorator'
```
Comment (severinbeauvais) [inline src/mixins/filing-mixin.ts:1]: No longer needed.

**Review by lewischenstudio — APPROVED** [review]
Looks good to me.

**Review by lambert-alex — APPROVED** [review]
LGTM

**Pull request merged**